external-api/src/repositories/estore/estore-repo.ts

```diff
@@ -0,0 +1,86 @@
+import { InsertOneResult, ObjectId, UpdateFilter, UpdateResult, WithId, WithoutId } from 'mongodb';
+import { MONGO_DB_COLLECTIONS } from '@ukef/dtfs2-common';
+import { getCollection } from '../../database';
+
+/**
+ * Finds a document in the CRON_JOB_LOGS collection based on the provided deal ID.
+ *
+ * @param {ObjectId} dealId - The unique identifier for the deal.
+ *
+ * @returns {Promise<WithId<ObjectId> | WithoutId<unknown> | null>} - A promise that resolves to the found document, or `null` if no document is found.
+ *
+ * @throws {Error} - Throws an error if the find operation fails.
+ *
+ * @example
+ * const dealId = new ObjectId('507f1f77bcf86cd799439011');
+ *
+ * find(dealId)
+ *   .then((document) => console.log('Document found', document))
+ *   .catch((error) => console.error('Find operation failed', error));
+ */
+export const find = async (dealId: ObjectId): Promise<WithId<ObjectId> | WithoutId<unknown> | null> => {
```

Comment:
`WithId<ObjectId>` is not the type you're expecting I think. `WithId` itself produces a type which contains the `_id: ObjectId` field. `ObjectId` itself has it's own type definition, however, so the type you've written is actually `ObjectId & { _id: ObjectId }` which I don't think you want.

Similarly, if `find` returns something, it will always contain the `_id: ObjectId` property by the nature of mongo, so the `WithoutId<unknown>` return type isn't quite right

Reply:
Have updated to `Promise<WithId<object> | null>`